src/components/_tests/ContextBar.test.js

```diff
@@ -0,0 +1,392 @@
+import '@testing-library/jest-dom'
+
+import { fireEvent, render, screen } from '@testing-library/react'
+import { div, h } from 'react-hyperscript-helpers'
+import { tools } from 'src/components/notebook-utils'
+import { MenuTrigger } from 'src/components/PopupTrigger'
+import { Ajax } from 'src/libs/ajax'
+import { CloudEnvironmentModal } from 'src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal'
+
+import { ContextBar } from '../ContextBar'
+
+
+// Mocking for terminalLaunchLink using Nav.getLink
+jest.mock('src/libs/nav', () => {
+  const originalModule = jest.requireActual('src/libs/nav')
+
+  return {
+    ...originalModule,
+    getPath: jest.fn(() => '/test/'),
+    getLink: jest.fn(() => '/')
+  }
+})
+
+// Mocking PopupTrigger to avoid test environment issues with React Portal's requirement to use
+// DOM measure services which are not available in jest environment
+jest.mock('src/components/PopupTrigger', () => {
+  const originalModule = jest.requireActual('src/components/PopupTrigger')
+  return {
+    ...originalModule,
+    MenuTrigger: jest.fn()
+  }
+})
+
+jest.mock('src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal', () => {
+  const originalModule = jest.requireActual('src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal')
+  return {
+    ...originalModule,
+    CloudEnvironmentModal: jest.fn()
+  }
+})
+
+jest.mock('src/libs/ajax')
+beforeEach(() => {
+  MenuTrigger.mockImplementation(({ content }) => { return div([content]) })
+  CloudEnvironmentModal.mockImplementation(({ isOpen, filterForTool, onSuccess, onDismiss, ...props }) => {
+    return isOpen ? div([
+      'Cloud Environment Details',
+      div(filterForTool),
+      div({ label: 'Success Button', onClick: () => onSuccess() }, 'SuccessButton'),
+      div({ label: 'Success Button', onClick: () => onDismiss() }, 'DismissButton')
+    ]) : div([])
+  })
+  Ajax.mockImplementation(() => {
+    return {
+      Metrics: {
+        captureEvent: () => {}
+      },
+      Runtimes: {
+        runtime: () => {
+          return {
+            start: jest.fn()
+          }
+        }
+      }
+    }
+  })
+})
+
+//Note - These constants are copied from src/libs/runtime-utils.test.js
+const galaxyRunning = {
```

Comment:
can we make a common test data source and override the necessary fields?

Reply:
I discussed that with Miguel, and I believe our agreement was that for the time being we'll keep the data in this file.

Reply:
I discussed this with Miguel. I am a strong believer we should form up a common data file for runtimes/disks. A lot of tests will need a generic 'running galaxy' runtime. This is a pattern we use everywhere for leo tests. 

Especially since this data is duplicated almost verbatim, and there will be thousands of lines of duplicated data otherwise.

Reply:
If you'd like, you can keep it as-is, and I can refactor.

Reply:
My thoughts on this topic are certainly colored by working in Typescript codebases.

With types on these data objects, I do prefer a bias towards locally specified mock objects for all but auto-generated large datasets... for a few reasons:
1) when the shape of an object changes, its great to get the signal of the impact by seeing and fixing the errors across the various code and test files.
2) the signal in #1 is a good code smell to maybe address DRY in the code structure to cut down on overly broad propagation/repetition of business logic, etc.
3) with Typescript, individual test files can use Partial<T> to reduce the verbosity of certain mock objects to just the essential properties/sub-objects that drive the given code-under-test.
4) #3 can also be a code smell that smaller interfaces should be defined to reduce the object-shape dependency surfaces of the code-under-test, which improves code reusability.
5) avoids cross-talk and dependency tentacles between test files.

Having said all that, this is not a hill I would die on.  What I would recommend though, is to refrain from any test data unification until AFTER the components folder is better organized to not be so flat, allowing team/functional areas to have a more natural place to put their specific test data.  Also, having Typescript in place first would help with cases where teams decide to use base mock objects with ...spread overrides, to assist in type correctness and discoverability.